torch/_inductor/scheduler.py

```diff
@@ -1869,6 +1903,20 @@ def _init(self, nodes: List[ir.Operation]) -> None:
             }
         )
 
+    def get_donated_buffer(self) -> Dict[str, SchedulerDonatedBuffer]:
+        name_to_donated_buf = {}
+        for node in self.nodes:
```

Comment:
Do we need to iterate over every node and every read just to get the Donated Buffers ?